src/uct/tcp/tcp_iface.c

```diff
@@ -599,62 +599,45 @@ static ucs_status_t uct_tcp_query_tl_resources(uct_md_h md,
                                                uct_tl_resource_desc_t **resource_p,
                                                unsigned *num_resources_p)
 {
-    uct_tl_resource_desc_t *resources, *tmp, *resource;
-    static const char *netdev_dir = "/sys/class/net";
-    struct dirent *entry;
-    unsigned num_resources;
+    uct_tl_resource_desc_t *resources, *resource, tmp_rsc;
+    unsigned num_resources, i;
+    char *dev_names;
+    char *tmp_dev_name;
     ucs_status_t status;
-    DIR *dir;
-
-    dir = opendir(netdev_dir);
-    if (dir == NULL) {
-        ucs_error("opendir(%s) failed: %m", netdev_dir);
-        status = UCS_ERR_IO_ERROR;
-        goto out;
-    }
 
     resources     = NULL;
     num_resources = 0;
-    for (;;) {
-        errno = 0;
-        entry = readdir(dir);
-        if (entry == NULL) {
-            if (errno != 0) {
-                ucs_error("readdir(%s) failed: %m", netdev_dir);
-                ucs_free(resources);
-                status = UCS_ERR_IO_ERROR;
-                goto out_closedir;
-            }
-            break; /* no more items */
-        }
 
-        if (!ucs_netif_is_active(entry->d_name)) {
-            continue;
-        }
+    status = ucs_sockaddr_get_dev_names(&num_resources, &dev_names, 
+                                        sizeof(tmp_rsc.dev_name));
+    if (UCS_OK != status) {
+        ucs_error("sockcm unable to find a sockets-capable device");
```

Comment:
I posted a comment to add in an error path inside the function
here we can remove `ucs_error(..)` and just do `goto out;`